pokemongo_bot/walkers/polyline_generator.py

```diff
@@ -104,30 +117,38 @@ def __init__(self, origin, destination, google_map_api_key=None):
                                          x['elevation']) for x in
                                         self._elevation_response['results'])
 
-    def _get_directions_points(self):
+    # Returns a list of points along the route. Each point consists of a tuple
+    # ((lat,long),speed)
+    def _get_directions_points_and_speeds(self):
         points = []
-        for point in self._directions_encoded_points:
-            points += polyline.decode(point)
-        return [x for n, x in enumerate(points) if x not in points[:n]]
+        speeds = []
+        for point in zip(self._directions_encoded_points, self._directions_speeds):
+            new_points = polyline.decode(point[0])
+            points += new_points
+            speeds += [ point[1] * uniform(0.9, 1.1) for p in new_points ]
```

Comment:
if you are modifying them with +/- 10% you are kinda removing canceling the variance anyway....